sdk/eventhub/azure-eventhub/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,228 @@
+  - [Configuring logback](#configuring-logback)
+  - [Enable AMQP transport logging](#enable-amqp-transport-logging)
+  - [Reduce logging](#reduce-logging)
+- [Troubleshoot EventProducerAsyncClient/EventProducerClient issues](#troubleshoot-eventproducerasyncclienteventproducerclient-issues)
+  - [Cannot set multiple partition keys for events in EventDataBatch](#cannot-set-multiple-partition-keys-for-events-in-eventdatabatch)
+  - [Setting partition key on EventData is not set in Kafka consumer](#setting-partition-key-on-eventdata-is-not-set-in-kafka-consumer)
+- [Troubleshoot EventProcessorClient issues](#troubleshoot-eventprocessorclient-issues)
+  - [412 precondition failures when using an event processor](#412-precondition-failures-when-using-an-event-processor)
+  - [Partition ownership changes frequently](#partition-ownership-changes-frequently)
+  - ["...current receiver '<RECEIVER_NAME>' with epoch '0' is getting disconnected"](#current-receiver-receiver_name-with-epoch-0-is-getting-disconnected)
+  - [High CPU usage](#high-cpu-usage)
+  - [Processor client stops receiving](#processor-client-stops-receiving)
+  - [Migrate from legacy to new client library](#migrate-from-legacy-to-new-client-library)
+- [Get additional help](#get-additional-help)
+  - [Filing GitHub issues](#filing-github-issues)
+
+## Handle Event Hubs exceptions
+
+All Event Hubs exceptions are wrapped in an [EventHubError][EventHubError].  They often have an underlying AMQP error code which specifies whether an error should be retried.  For retryable errors (ie. `amqp:connection:forced` or `amqp:link:detach-forced`), the client libraries will attempt to recover from these errors based on the [retry options][AmqpRetryOptions] specified when instantiating the client.  To configure retry options, follow the sample [Client Creation][ClientCreation].  If the error is non-retryable, there is some configuration issue that needs to be resolved.
+
+The recommended way to solve the specific exception the AMQP exception represents is to follow the
+[Event Hubs Messaging Exceptions][EventHubsMessagingExceptions] guidance.
+
+### Find relevant information in exception messages
+
+An [EventHubError][EventHubError] contains three fields which describe the error.
+
+* **message**: The underlying AMQP error message. A description of the errors can be found in the [Exceptions module][ExceptionModule] or the [OASIS AMQP 1.0 spec][AmqpSpec].
+* **error**: The error condition if available.
+* **details**: The error details, if included in the service response
+  
+### Commonly encountered exceptions
+
+#### `amqp:connection:forced` and `amqp:link:detach-forced`
+
+When the connection to Event Hubs is idle, the service will disconnect the client after some time.  This is not a problem as the clients will re-establish a connection when a service operation is requested.  More information can be found in the [AMQP troubleshooting documentation][AmqpTroubleshooting].
```

Comment:
The service will boot idle connections after 20 minutes, and there's no way to control that from the client.   Since this is service behavior and the duration is not guaranteed, we chose to be deliberately vague.